File: ch3/3-1.cpp
/*
Program 1 Chapter 3 
Name: Three in One
Descrption: Desribe how you could use a single array to implemented three stacks 
Runtime: 
Date: 4/3/2019
*/

#include <iostream>
#include <array>
using namespace std;




//take in the array number and the item to push 
void push(int arrayNum, int value)
{
    cout << "hi" << endl;
}



int main()
{
    
    array<int,15>myarray; 
    for (int i=0; i<15; i++)
    {
        myarray[i] = 0;
    }

    int size = myarray.size();
    int numArrays = size/3;
    cout << "size of segments: " << numArrays << endl;
    int arr1 = numArrays;
    int arr2 = numArrays*2;
    int arr3 = numArrays*3;
    cout << arr1 << " " << arr2 << " " << arr3 << endl;
    myarray[0] = 1;


    for (int i=0; i<15; i++)
    {
        cout << myarray[i] << " " <<  endl;
    }
    return 0;
}

File: ch3/3-1.py
#Program 1 Chapter 3 
#Name: Three in one 
#Runtime: O(1)
#Date: 4/3/2019

#iterators for each "separate" array
first = 0
second = 5
third = 10

def addToArr(array, arrayNum, val):

    #global so that we can increment iterators 
    global first
    global second
    global third

    # 1/3 of full array
    if arrayNum == 1:
        if first <= 4:
            array[first] = val
            first+=1
        else:
            print("not enough space!")

    # 2/3 of full array
    elif arrayNum == 2:
        if second <= 9:
            array[second] = val
            second+=1
        else:
            print("not enough space!")

    # 3/3 of full array
    elif arrayNum ==3:
        if third <= 14:
            array[third] = val
            print(str(third))
            third+=1
        else:
            print("not enough space!")


def main():

    array = [None]*15
    
    #adding to first "array"
    addToArr(array, 1,100)
    print(array)
    addToArr(array, 1,100)
    print(array)
    addToArr(array, 1,100)
    print(array)
    addToArr(array, 1,100)
    print(array)
    addToArr(array, 1,100)
    print(array)
    addToArr(array, 1,100)
    print(array)

    addToArr(array, 2,200)
    print(array)
    addToArr(array, 2,200)
    print(array)
    addToArr(array, 2,200)
    print(array)
    addToArr(array, 2,200)
    print(array)
    addToArr(array, 2,200)
    print(array)
    addToArr(array, 2,200)
    print(array)

    addToArr(array, 3,300)
    print(array)
    addToArr(array, 3,300)
    print(array)
    addToArr(array, 3,300)
    print(array)
    addToArr(array, 3,300)
    print(array)
    addToArr(array, 3,300)
    print(array)
    addToArr(array, 3,300)
    print(array)

if __name__ == "__main__":
    main()








